projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -1302,5 +1303,9 @@ export function getLineFromTSFile(
 export function getServerTsPath(host: Tree): string | undefined {
   const projectName = getDefaultProjectNameFromWorkspace(host);
   const angularJson = getAngularJsonFile(host);
-  return angularJson.projects[projectName].architect?.server?.options?.main;
+
+  return (
+    angularJson.projects[projectName].architect?.server?.options?.main ||
+    (angularJson.projects[projectName].architect as any)?.build.options.server
+  );
```

Comment:
Thank you for the explanation. Do these changes occur also in applications migrated to Angular 17? Or only in fresh Angular 17 apps? I'm asking, because I couldn't find such a property in the `angular.json` file in the migrated application.

If it's only the fresh app case, maybe we don't need to worry about it right now as the `checkIfSsrIsUsed` function is used only during the migration.

Reply:
yes, you are right. I was running these schematics without migrations and misunderstood it as a bug.

Reply:
I've removed this part